crates/rerun_c/src/lib.rs

```diff
@@ -579,6 +579,8 @@ fn rr_log_impl(
     data_row: CDataRow,
     inject_time: bool,
 ) -> Result<(), CError> {
+    let row_id = re_sdk::log::RowId::new(); // Create row-id as early as possible. It has a timestamp and is used to estimate e2e latency.
```

Comment:
This is a bit of a bigger lift, but we'd get a waaay more accurate number if we create the rowid on the C++ side. Benchmarks indicate that reaching this function can take a significant amount of time.

I propose that `rr_log` takes a `rr_row_id` which was previously obtained using `rr_create_row_id`. That way, we the C++ recording stream can obtain the row id before doing all the arrow serialization and arrow-FFI boundary things!

Reply:
I added a TODO